### DISCUSSION TITLE
Welcome to table-sort-js Discussions!

### DISCUSSION THREAD

**Discussion by kyle-wannacott** [LeeWannacott/table-sort-js#31] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.

**Comment by ADVitapur** [comment]
Any ideas how to add subtables to sorting?

for example, i have a table, and if you click on a button it shows another table.
If I use the table-sort, the subtables get messed up.
I can probably look at the code and fix it, but I want to ask first before I delve into it

**Reply by kyle-wannacott** [reply]
> Any ideas how to add sub-tables to sorting?

Not off the top of my head; can't say I have thought about it. Are sub-tables like you click an option from a drop-down and it changes to another table? :thinking: 

Would you be able to post a screenshot of what you are trying to do; or/and maybe some code?

**Comment by dca00** [comment]
Just landed here while looking for an alternative to Sorttables.js, which would not break Tabber's tabs, and this solution worked OOB. Only had to figure out how to get arrows, by debugging. Worth mentioning that class "table-arrows" should be added for that.
Thanks for a great piece of code and keep up good work!

**Reply by kyle-wannacott** [reply]
Hey @dca00, 

> Which would not break Tabber's tabs

No idea what "Tabber's tabs is", but good that it works with table-sort-js.

> Just landed here while looking for an alternative to Sorttables.js

I think Sorttables.js was written a long time ago; has like web-dings. 

> Only had to figure out how to get arrows, by debugging. Worth mentioning that class "table-arrows" should be added for that.

Are you saying that "table-arrows" was useful for debugging?. The implementation of "table-arrows" isn't that great.

> Thanks for a great piece of code and keep up good work!

Thanks, honestly I haven't really worked on this project in a while. I keep starting projects and not finishing them :sweat_smile: ; although is software ever finished :shrug:

**Reply by dca00** [reply]
I am saying that in order to find out that arrows can be added by adding table-arrows class I had to debug your code. I did not find a mention of table-arrows in the documentation.

**Reply by kyle-wannacott** [reply]
@dca00 
https://github.com/LeeWannacott/table-sort-js#readme
![image](https://user-images.githubusercontent.com/49783296/188565979-23902334-f388-4bb2-a278-f2feae805863.png)https://leewannacott.github.io/table-sort-js/docs/classes.html
![image](https://user-images.githubusercontent.com/49783296/188565496-0fb659f1-0c17-465c-b01b-dee6955cea7f.png)
![image](https://user-images.githubusercontent.com/49783296/188565684-5845358c-5c96-49c5-bb0c-57df2a926160.png)

**Comment by Osiaris** [comment]
First of all, this is a great and easy to understand table sorter, thank you so much for building it! I'm building a React app and am having trouble sorting decimals. It seems to be ordering by number of decimals rather than the value, i am adding a picture of an ordered list by lowest to highest. 0.3 is a higher number than 0.11 but 0.3 is higher. Is this a bugg, am i missunderstanding or do you think i am doing something wrong?

![bild](https://user-images.githubusercontent.com/94265685/190397837-c87c737e-108a-445f-9599-6785b5c8d614.png)

**Reply by Osiaris** [reply]
I have actually now seen the same issue on your webside, i am adding a picture where i have ordered the list on "release version", 3.9.12 is a higher version number than 3.18.12 yet it is lower in the list. 

![bild](https://user-images.githubusercontent.com/94265685/190402097-0253ef10-cf6e-418c-a8b0-1419e27ce448.png)

**Reply by Osiaris** [reply]
Thank you for a quick reply! I will attempt to open an issue, my image is just a small screenshot of a couple of dynamically created tables the user can create 😂 but i will attempt to build a workaround. Thank you!

**Reply by strager** [reply]
> 3.9.12 is a higher version number than 3.18.12 yet it is lower in the list.

I think you are incorrect. 3.18.12 is a higher/newer version number than 3.9.12.

**Comment by goodchiang** [comment]
I'm searching a way to reset the table to the initial state, but I can't find any solution.
My idea is that when clicking on the thead 3 times, and the table columns is back to initial unsorting state.
Is that any solution for this idea?

**Reply by kyle-wannacott** [reply]
@goodchiang Sorry, there is currently no way to get back to the state the column was originally in, other than refreshing your browser 😅. The original order currently isn't cached in any way. You can open a GitHub issue if you want...

**Reply by goodchiang** [reply]
@LeeWannacott I think I found a solution as there is already one column to record the original sorting index, so I just tried to sort that specific column when it's the third time clicking. And I guess it works well for my case... (hope).
